crawler4j/src/main/java/edu/uci/ics/crawler4j/crawler/CrawlController.java

```diff
@@ -155,6 +155,21 @@ public Parser getParser() {
         T newInstance() throws Exception;
     }
 
+    private static class SingleInstanceFactory<T extends WebCrawler>
```

Comment:
How is this related to the rest of the patch?

Reply:
I missed to push test. See PR.